test/application.js

```diff
@@ -412,9 +412,12 @@ describe('app.respond', function(){
         this.status = 200;
         res.setHeader("Content-Type", "text/html")
         res.write('Hello');
-        setTimeout(function(){
-          res.end("Goodbye")
-        }, 0);
+        yield new Promise(function(resolve){
+          setTimeout(function(){
+            res.end("Goodbye");
+            resolve();
+          }, 0);
+        });
```

Comment:
A valid point in that this might be a breaking change. However, this code relies on the fact that `respond` will do nothing if headers have been sent, which is exactly what this change tries to stop. We could have a variable that would enable this new functionality, but that seems like a hack.